viewing_party/party.py

```diff
@@ -1,23 +1,147 @@
 # ------------- WAVE 1 --------------------
 
+# returns a dict 
 def create_movie(title, genre, rating):
-    pass
+    movie = {}
+    if title and genre and rating:
+        movie["title"] = title
+        movie["genre"] = genre
+        movie["rating"] = rating
+        return movie
+    else:
+        return None
+# add and returns the movie to "watched" list inside of userdata
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
 
+# add and returns movie to "watchlist" list inside of user_data
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+# If title is a movie in user's watchlist it will remove it from watchlist and add it to watched list
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            add_to_watched(user_data, movie)
+            print(movie)
+    i = 0
+    while i < len(user_data["watchlist"]):

```

Comment:
I'm actually not quite sure that I am following how this is working :) It's a clever set up, but.. Note, while this may work here(?), it's considered dangerous practice to modify the content of a data structure that you're looping over.  Especially in more complex code, this can open you up to complex errors. A simple alternative here could be to save a copy of what you get from movie in user_data["watchlist"] and iterate over that, so that as you modify user_data["watchlist"], there's no risk.